### PR TITLE
Feature/text toolbar

### PR DESCRIPTION
**Pull request by SukranSari**

### CODE REVIEW

**Review by Roland-Hufnagel — CHANGES_REQUESTED** [review]
Hi team. Nice editor 👍. Works fine. I left some comments to improve the code a bit.

File: pages/index.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ export default function HomePage({
   isFormOpen,
 }) {
   const [searchInput, setSearchInput] = useState("");
+  const [value, setValue] = useState("");
```
[the remark this answers is not in the document]
Comment (SukranSari) [reply pages/index.js]: It is removed.

File: package.json [hunk 2]
```diff
@@ -17,18 +17,22 @@
...
     "eslint-config-next": "13.4.8",
     "formidable": "^3.5.1",
+    "html-react-parser": "^5.1.12",
     "install": "^0.13.0",
     "nanoid": "^5.0.7",
     "next": "13.4.8",
     "next-images": "^1.8.5",
     "npm": "^10.8.2",
+    "primereact": "^10.8.2",
+    "quill": "^2.0.2",
```
[the remark this answers is not in the document]
Comment (SukranSari) [reply package.json:27]: It is removed

File: components/Note/index.js [hunk 3]
```diff
@@ -4,13 +4,18 @@ import styled from "styled-components";
 import useLocalStorageState from "use-local-storage-state";
 import { SlNote } from "react-icons/sl";
 import { TfiCheck } from "react-icons/tfi";
+import { Editor } from "primereact/editor";
+import parse from "html-react-parser";
 
 const Note = ({ project }) => {
```
[the remark this answers is not in the document]
Comment (SukranSari) [reply components/Note/index.js:10]: It is removed

File: components/Note/index.js [hunk 4]
```diff
@@ -4,13 +4,18 @@ import styled from "styled-components";
...
   const router = useRouter();
   const { id } = router.query;
 
   const [currentNote, setCurrentNote] = useLocalStorageState(`note-${id}`, "");
-  const [isEditMode, setIsEditMode] = useState(false);
+  const [isEditMode, setIsEditMode] = useLocalStorageState(
```
[the remark this answers is not in the document]
Comment (SukranSari) [reply components/Note/index.js:15]: I have forgotten to delete the useState. We have prefered the LocalStorage to get a overwiev about the changes

File: components/Note/index.js [hunk 5]
```diff
@@ -19,28 +24,38 @@ const Note = ({ project }) => {
...
       {isEditMode ? (
-        <StyledTextarea
-          ref={textareaFocus}
+        <Editor
           value={currentNote}
-          onChange={handleInputChange}
-          rows="5"
-          cols="50"
-          placeholder="Enter your notes here..."
+          onTextChange={(e) => handleTextChange(e)}
```
[the remark this answers is not in the document]
Comment (SukranSari) [reply components/Note/index.js:52]: ok

File: components/Note/index.js [hunk 6]
```diff
@@ -19,28 +24,38 @@ const Note = ({ project }) => {
     }
   }, [isEditMode]);
 
-  const handleInputChange = (event) => {
-    setCurrentNote(event.target.value);
+  const handleTextChange = (event) => {
+    setCurrentNote(event.htmlValue);
+    console.log(event.textValue);
```
[the remark this answers is not in the document]
Comment (SukranSari) [reply components/Note/index.js:29]: ok

File: components/Note/index.js [hunk 7]
```diff
@@ -4,13 +4,18 @@ import styled from "styled-components";
...
   const router = useRouter();
   const { id } = router.query;
 
   const [currentNote, setCurrentNote] = useLocalStorageState(`note-${id}`, "");
-  const [isEditMode, setIsEditMode] = useState(false);
+  const [isEditMode, setIsEditMode] = useLocalStorageState(
+    `editMode-${id}`,
+    false
+  );
   const textareaFocus = useRef(null);
```
[the remark this answers is not in the document]
Comment (SukranSari) [reply components/Note/index.js:19]: It is removed

File: components/Note/index.js [hunk 8]
```diff
@@ -19,28 +24,38 @@ const Note = ({ project }) => {
...
   const toggleEditMode = () => {
     setIsEditMode(!isEditMode);
   };
 
+  const customToolbar = (
+    <span className="ql-formats">
+      <button className="ql-bold" />
+      <button className="ql-italic" />
+      <button className="ql-underline" />
+      <button className="ql-list" value="ordered" />
```
[the remark this answers is not in the document]
Comment (SukranSari) [reply components/Note/index.js:32]: ok

**Pull request merged**